src/Lib/ImageTransformInterface.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * @category Proffer
+ * @package ImageTransformInterface.php
+ *
+ * @author David Yell <neon1024@gmail.com>
+ * @when 23/03/15
+ *
+ */
+
+namespace Proffer\Lib;
+
+use Cake\ORM\Table;
+
+interface ImageTransformInterface
+{
+
+    /**
+     * @param Table $table Instance of the current table
+     * @param ProfferPathInterface $path Instance of the current path class
+     */
+    public function __construct(Table $table, ProfferPathInterface $path);
```

Comment:
That's a shame you have to add the constructor here because of the dependency injection implementation, unless this is a Cake thing I'm not aware of? If I need to create a concrete class that implements ImageTransformInterface I'm forced to only have a Table and ProfferPathInterface